pkg/cableengine/ipsec/strongswan.go

```diff
@@ -291,15 +312,52 @@ func (i *strongSwan) loadSharedKey(endpoint types.SubmarinerEndpoint, client *go
 	return nil
 }
 
-func runCharon(debug bool, logFile string) error {
+const charonConfTemplate = `
+	charon {
+		port = {{.ipSecIKEPort}}
+		port_nat_t = {{.ipSecNATTPort}}
+		make_before_break = yes
+		ignore_acquire_ts = yes
+	}
+`
+
+func (i *strongSwan) writeCharonConfig(path string) {
+	err := os.Remove(path)
+	if err != nil {
+		klog.Infof("Error deleting %s", path)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		klog.Fatalf("Error creating %s: %s", path, err)
+	}
+	i.renderCharonConfigTemplate(f)
+}
+
+func (i *strongSwan) renderCharonConfigTemplate(f io.Writer) {
+	t, err := template.New("charon.conf").Parse(charonConfTemplate)
+	if err != nil {
+		klog.Fatalf("Error creating template for charon.conf")
+	}
+	if err = t.Execute(f, map[string]string{
+		"ipSecIKEPort":  i.ipSecIKEPort,
+		"ipSecNATTPort": i.ipSecNATTPort}); err != nil {
```

Comment:
(nit) it's a long if. I'd handle the err in a different line

Reply:
May be it will look better, I'll try, thanks.